### DISCUSSION TITLE
Wear OS watch battery drain when using xDrip data in complication

### DISCUSSION THREAD

**Discussion by ojalaj** [NightscoutFoundation/xDrip#2971] [Q&A]
Xdrip version e50dfdf-2023.05.03 both on phone and watch (Samsung Galaxy Watch 4). I'm using xDrip on watch only as complication in G-Watch Wear watchface (since I need to follow glucose data from two sources (LibreLink follower)).

I have experienced similar increased battery drain than reported earlier (e.g. see #2026 ). Watch restarts don't help and watch reset I don't see as favourable option, at least since it is not reported to be as a cure for this issue. Also, I wouldn't like to add another layer of apps for this, as suggested as a workaround in #2026, at least not until some experts on xDrip have addressed this issue.

I would be very grateful for some help on this - thank you.

**Comment by Navid200** [comment]
I hope someone who is interested in WearOS, who has experience and expertise, will join our team to provide support.
Ideally, they will move the WearOS portion of xDrip out into a companion app.
Something similar happened to MiBand.  It is now supported by someone with a lot of experience with MiBand.
In my opinion, that decentralization is a natural growth path for the project.

Currently, any change made to xDrip will need to go through a lot of testing because it could affect unintended components.
This results in a slower pace of change.

If a companion app handles WearOS, it can grow and change with no need to change xDrip.
There will have to be an interface added to xDrip.  But, after that is done, xDrip will not need to change for every single change required to be made to WearOS.

The last time a change was made to xDrip to address WearOS was a very long time ago.  
I personally have no experience with WearOS.  Otherwise, I would have done it.

Please note that is not an official xDrip statement.  It is only my personal opinion and I can be corrected.

**Reply by ojalaj** [reply]
@Navid200 Thank you for your reply. I understand the points raised. As I don't have skills to contribute, we'll need to wait until someone perhaps volunteers. You mentioned that last time WearOS related change was made was long time ago and in #2026 some users report that downgrading xDrip to certain version, this issue is non-existent - would it be worth double-checking the changes made around those versions?

**Reply by Navid200** [reply]
It's always useful to know if there is an unintended xDrip change of behavior.
So, if a user can install two consecutive versions of xDrip and show different behavior, where the change was not intended, it will make life much easier for developers.

**Reply by ojalaj** [reply]
At least on #2026 other users reported "I continue to stay on 2022.01.31 as any newer version eats battery like crazy."

**Reply by Navid200** [reply]
The release after 2022.01.31 is 2022.02.09
The changes made in 2202.02.09 are some string translations, Dex share user id fix (https://github.com/NightscoutFoundation/xDrip/pull/1978), and more logging options for Libre 2 (https://github.com/NightscoutFoundation/xDrip/pull/1961).
There is nothing related to WearOS.

But, you see the main problem here?  I am not a WearOS user.  A WearOS user may be able to find something that is the cause looking at the differences between the two releases, I cannot.  
We don't have an active developer who is using WeasrOS.

**Reply by ojalaj** [reply]
Yes, I understand. I was just trying to help identifying version from which the observed behavior has been reported to start. Hopefully someone who would understand code related to WearOS could jump in. I apologize for bothering you.
